example-app/src/main/java/com/adyen/checkout/example/di/NetworkModule.kt

```diff
@@ -15,27 +15,42 @@ import com.adyen.checkout.example.data.api.CheckoutApiService
 import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import java.security.KeyStore
+import java.util.*
+import javax.inject.Singleton
+import javax.net.ssl.TrustManagerFactory
+import javax.net.ssl.X509TrustManager
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
-import org.koin.dsl.module
+import retrofit2.Converter
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
-import java.security.KeyStore
-import java.util.Arrays
-import javax.net.ssl.TrustManagerFactory
-import javax.net.ssl.X509TrustManager
 
-val networkModule = module {
+@Module
+@InstallIn(SingletonComponent::class)
+object NetworkModule {
 
-    fun provideHttpClient(): OkHttpClient {
-        val builder = OkHttpClient.Builder().let {
+    private val BASE_URL = if (CheckoutApiService.isRealUrlAvailable())
+        BuildConfig.MERCHANT_SERVER_URL
+    else
+        "http://myserver.com/my/endpoint/"
+
+    @Singleton
```

Comment:
Do all these methods have to be singletons?

Reply:
Not really, I think it's good practice for the Retrofit instance to be a singleton that's reused by the repositories, and the repos are also singletons usually, so idk if there would be any benefit of trying to have them cleared from memory at some point.